src/smk/tool/geomark/tool-geomark.js

```diff
@@ -322,16 +338,16 @@ include.module( 'tool-geomark', [
                     self.showPrompt = true;
                 },
                 'toggle-geomark': function(idObj) {
-                    const geomark = self.getGeomarkById(idObj.id);
-                    if (!geomark) {
+                    const geomarkInfo = self.getGeomarkById(idObj.id);
+                    if (!geomarkInfo) {
                         return;
                     }
-                    if (smk.$viewer.map.hasLayer(geomark.drawingLayer)) {
-                        smk.$viewer.map.removeLayer(geomark.drawingLayer);
+                    if (geomarkInfo.isVisible) {
+                        smk.$viewer.map.getPane(geomarkInfo.layerUUID).style.display = 'none';
```

Comment:
is getPane() guranteed to return an object? Even if the UUID isn't found? If so then this is probably safe. If not, you should add a defensive null check before accessing style.display.

Reply:
Good catch! If the UUID isn't found, it returns undefined. I'll add a check in another commit.